solr/solrj/src/java/org/apache/solr/common/util/XML.java

```diff
@@ -101,8 +101,12 @@ public final static void writeXML(Writer out, String tag, String val) throws IOE
     }
   }
 
-  /** does NOT escape character data in val, must already be valid XML */
   public final static void writeUnescapedXML(Writer out, String tag, String val, Object... attrs) throws IOException {
+    writeUnescapedXML(out, tag, (writer1) -> writer1.write(val), attrs);
+  }
+
+  /** does NOT escape character data in val, must already be valid XML */
+  public final static void writeUnescapedXML(Writer out, String tag, Writable val, Object... attrs) throws IOException {
```

Comment:
The method name should not include "unescaped".  Wether it's escaped or not is up to the Writable.  "val" variable name should be valWritable or some-such.  The javadoc comment about what the method does belongs on the method above which you added.